### DISCUSSION TITLE
Trying to get to close application but not closing thread.

### DISCUSSION THREAD

**Discussion by lukshsahitia** [TolikPylypchuk/SharpHook#23] [Q&A] [answered]
Hi, I'm trying to use your code for my maui spreadhsheet project but unfortunately it will not run as I intend it to. The code works fine and everything while the application is running but when I try to close the window, the application or thread still seems to be running so I believe the hook.dispose() code is not disposing the key handlers properly.

This is my code, please let me know if I am doing something wrong.
[KeyPressed.txt](https://github.com/TolikPylypchuk/SharpHook/files/9833752/KeyPressed.txt)

**Comment by TolikPylypchuk** [comment, marked as answer]
Hi! Thanks for using SharpHook!

As far as I can see the problem is in this code:

```c#
await hook.RunAsync();
hook.Dispose();
```

The `Dispose` method is never actually going to be called. In the code above you need to stop awaiting `RunAsync` to dispose, but `RunAsync` will finish only after disposing. You need to call `Dispose` in another place - some code that gets called when the application is about to stop, like a window closed event handler. I don't have any experience with MAUI, so I can't help with the specifics.

Actually, there's an issue in SharpHook itself. The hook runs in a separate thread - that thread should be a background thread, and it isn't. Thanks for the catch, I'll fix that in the next version, but for now you should call `Dispose` when the application exists.

**Comment by lukshsahitia** [comment]
Got it thanks boss!
